src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -751,6 +752,65 @@ class RandomDateGenerator : public Generator<bsoncxx::types::b_date> {
     const UniqueGenerator<int64_t> _maxGen;
 };
 
+class CycleGenerator : public Appendable {
+
+public:
+    CycleGenerator(const Node& node,
+                    GeneratorArgs generatorArgs,
+                    std::map<std::string, Parser<UniqueAppendable>> parsers)
+        : CycleGenerator(
+            extract(node, "of", "^Cycle"),
+            generatorArgs,
+            parsers,
+            extract(node, "number", "^Cycle").to<int64_t>()) {}
+
+    CycleGenerator(const Node& node,
+                GeneratorArgs generatorArgs,
+                std::map<std::string, Parser<UniqueAppendable>> parsers,
+                int64_t size)
```

Comment:
it seems we call this `number`, `size`, and `_cacheSize`. This one is a hard thing to name. Consider just `_cycleNumber` to be consistent with the user-facing syntax.

Reply:
Good thought, I changed the user-facing syntax to use ofLength and updated the naming here to match in each spot